scalding-core/src/main/scala/com/twitter/scalding/serialization/PositionInputStream.scala

```diff
@@ -27,20 +27,23 @@ object PositionInputStream {
 
 class PositionInputStream(val wraps: InputStream) extends InputStream {
   private[this] var pos: Long = 0L
-  private[this] var markPos: Long = -1L
   def position: Long = pos
 
   override def available = wraps.available
+
   override def close() { wraps.close() }
+
   override def mark(limit: Int) {
-    wraps.mark(limit)
-    markPos = pos
+    sys.error("Mark and reset are not supported on this stream")
   }
-  override def markSupported: Boolean = wraps.markSupported
+
+  override val markSupported: Boolean = false
+
   override def read: Int = {
-    val count = wraps.read
-    if (count > 0) pos += 1
-    count
+    val result = wraps.read
```

Comment:
good catch here. I had some moment of confusion and thought, like read on bytes, that it was returning the count. I realized it was not later (because I had to write `pos += 1` not `pos += count`, but I failed fix the `== 0` case.